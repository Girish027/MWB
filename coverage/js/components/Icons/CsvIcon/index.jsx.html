
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for js/components/Icons/CsvIcon/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> / <a href="index.html">js/components/Icons/CsvIcon</a> index.jsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">60% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>3/5</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">0% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">60% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>3/5</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
&nbsp;
const CsvIcon = <span class="fstat-no" title="function not covered" >(p</span>rops) =&gt; {
  const {
    height, width, fill, ...otherProps
  } = <span class="cstat-no" title="statement not covered" >props;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >  return (</span>
    &lt;svg width={width} height={height} viewBox="0 0 49 49" {...otherProps}&gt;
      &lt;g fill={fill} fillRule="nonzero"&gt;
        &lt;path d="M48.487 22.1a.986.986 0 0 0-.053-.263c-.011-.032-.019-.063-.033-.094a.955.955 0 0 0-.193-.284l-.001-.001-5.707-5.693v-5.571c0-.022-.011-.041-.013-.063a1.023 1.023 0 0 0-.051-.256c-.011-.032-.019-.063-.034-.094a.994.994 0 0 0-.196-.292l-9-8.976a.97.97 0 0 0-.294-.196c-.03-.014-.06-.022-.091-.032a.94.94 0 0 0-.263-.052C32.539.23 32.521.22 32.5.22h-25c-.552 0-1 .447-1 .998v14.547L.793 21.457l-.002.002a.977.977 0 0 0-.192.284c-.014.03-.022.062-.033.094a.948.948 0 0 0-.053.263c-.002.022-.013.04-.013.062v18.95c0 .55.448.996 1 .996h5v4.987c0 .55.448.997 1 .997h34c.552 0 1-.446 1-.997v-4.987h5c.552 0 1-.446 1-.997v-18.95c0-.021-.011-.04-.013-.061zm-3.401-.936H42.5v-2.579l2.586 2.58zm-6-11.968H33.5v-5.57l5.586 5.57zM8.5 2.216h23v7.978c0 .55.448.997 1 .997h8V21.164h-32V2.215zm-2 16.37v2.578H3.914L6.5 18.585zm34 27.512h-32v-3.99h32v3.99zm6-5.984h-44V23.159h44v16.955z" /&gt;
        &lt;path d="M23.922 28.031c.147-.136.312-.234.493-.297.181-.062.368-.093.561-.093.669 0 1.224.265 1.666.797l1.122-1.458a3.164 3.164 0 0 0-1.215-.874c-.482-.197-1.028-.296-1.64-.296-.419 0-.833.07-1.241.211a3.145 3.145 0 0 0-1.097.654 3.276 3.276 0 0 0-.782 1.11c-.198.447-.298.981-.298 1.602 0 .498.065.924.195 1.28.13.357.306.667.527.933.221.266.476.494.765.687.289.192.598.367.927.525.521.26.952.543 1.292.848.34.305.51.718.51 1.238 0 .531-.142.943-.425 1.237-.283.295-.64.441-1.071.441-.385 0-.762-.09-1.131-.271a2.816 2.816 0 0 1-.944-.746l-1.105 1.492c.34.396.793.723 1.36.983.567.26 1.184.39 1.853.39.465 0 .907-.078 1.326-.237a3.077 3.077 0 0 0 1.097-.704 3.45 3.45 0 0 0 .748-1.144c.187-.452.281-.973.281-1.56 0-.509-.079-.95-.238-1.322a3.366 3.366 0 0 0-.612-.967 4.063 4.063 0 0 0-.833-.695 10.331 10.331 0 0 0-.901-.509c-.499-.248-.901-.511-1.207-.788-.306-.277-.459-.67-.459-1.178 0-.295.042-.549.128-.763.085-.215.201-.39.348-.526zM15.677 28.998c.204-.396.459-.71.765-.942a1.726 1.726 0 0 1 1.071-.347c.737 0 1.337.333 1.802 1l1.173-1.424a3.282 3.282 0 0 0-1.275-1.094c-.521-.254-1.128-.38-1.819-.38-.669 0-1.278.155-1.828.465a4.224 4.224 0 0 0-1.403 1.305c-.385.56-.683 1.227-.892 2.001a9.67 9.67 0 0 0-.314 2.535c0 .916.105 1.757.314 2.526.209.77.504 1.433.884 1.992.38.56.844.998 1.394 1.315.55.316 1.165.474 1.845.474.748 0 1.368-.146 1.861-.44a3.757 3.757 0 0 0 1.215-1.153l-1.173-1.374a2.76 2.76 0 0 1-.722.746 1.738 1.738 0 0 1-1.028.305c-.419 0-.785-.115-1.097-.348a2.674 2.674 0 0 1-.774-.94 5.103 5.103 0 0 1-.459-1.4 9.11 9.11 0 0 1-.153-1.703c0-.61.051-1.184.153-1.721a5.175 5.175 0 0 1 .46-1.398zM32.507 35.745h-.051l-2.006-9.63h-2.278l2.941 12.004h2.584l2.992-12.004h-2.176z" /&gt;
      &lt;/g&gt;
    &lt;/svg&gt;
  );
};
&nbsp;
CsvIcon.defaultProps = {
  height: 49,
  width: 49,
  fill: '#9B9B9B',
};
&nbsp;
CsvIcon.propTypes = {
  height: PropTypes.number,
  width: PropTypes.number,
  fill: PropTypes.string,
};
&nbsp;
export default CsvIcon;
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Aug 05 2021 15:06:58 GMT+0530 (IST)
            </div>
        </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    