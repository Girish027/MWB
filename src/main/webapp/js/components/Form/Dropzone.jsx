import React, { useMemo, useCallback } from 'react';
import { useDropzone } from 'react-dropzone';
import PropTypes from 'prop-types';
import Constants from 'constants/Constants';

export default function Dropzone(props) {
  const {
    accept,
    saveFile,
    rejectFile,
    multiple,
    disabled,
    shouldUpload,
    uploadHandler,
    icon: FileIcon,
    acceptedIcon: AcceptIcon,
    acceptStyle,
    activeStyle,
    rejectStyle,
    baseStyle,
    removeFile,
  } = props;

  const sizeRestrictions = {};
  if (props.maxSize) {
    sizeRestrictions.maxSize = props.maxSize;
  }
  if (props.minSize) {
    sizeRestrictions.minSize = props.minSize;
  }

  const status = {
    ready: 'ready',
    accepted: 'accepted',
    rejected: 'rejected',
    acceptedFiles: [],
  };

  const state = {
    status: status.ready,
  };

  const eventHandlers = {
    onDrop: useCallback((acceptedFiles) => {
      if (acceptedFiles.length) {
        state.status = status.accepted;
        state.acceptedFiles = removeFile ? [] : acceptedFiles;
        saveFile(removeFile ? [] : acceptedFiles);

        if (shouldUpload) {
          uploadHandler(removeFile ? [] : acceptedFiles);
        }
      }
    }, [saveFile, shouldUpload, state.acceptedFiles, state.status, status.accepted, uploadHandler]),

    onDropRejected: useCallback((rejectedFiles) => {
      state.status = status.rejected;
      state.rejectedFiles = rejectedFiles;
      rejectFile(rejectedFiles);
    }, [rejectFile]),
  };

  const {
    getRootProps,
    getInputProps,
    isDragActive,
    isDragAccept,
    isDragReject,
    acceptedFiles,
  } = useDropzone({
    ...eventHandlers,
    accept,
    multiple,
    disabled,
    preventDropOnDocument: true,
    ...sizeRestrictions,
  });

  const style = useMemo(() => ({
    ...baseStyle,
    ...(isDragActive ? activeStyle : {}),
    ...(isDragAccept ? acceptStyle : {}),
    ...(isDragReject ? rejectStyle : {}),
  }), [baseStyle, isDragActive, activeStyle, isDragAccept, acceptStyle, isDragReject, rejectStyle]);

  const renderDetails = (view) => {
    switch (view) {
    case status.ready:
      return (
        <div>
Drag and drop your file
          <div> or </div>
          <div style={{ cursor: 'pointer', color: '#004c97' }}> BROWSE</div>
        </div>
      );
    case status.accepted: {
      return Constants.DROPZONE.ACCEPCTED;
    }
    case status.rejected: {
      return Constants.DROPZONE.REJECTED(accept);
    }
    default: return null;
    }
  };


  let newAcceptedFiles = removeFile ? [] : acceptedFiles;
  return (
    <div {...getRootProps({ style })}>
      <input {...getInputProps()} />
      <div style={{ margin: 'auto' }} className="dropzone-content">
        {newAcceptedFiles.length > 0
          ? newAcceptedFiles.map((acceptedFile) => (
            <div style={{ display: 'inline-block' }} key={acceptedFile.name}>
              {AcceptIcon
                ? (
                  <div style={{ textAlign: 'center' }}>
                    <AcceptIcon height={48} width={48} fill="#004c97" />
                  </div>
                )
                : null
              }
              <div style={{ paddingTop: '10px', whiteSpace: 'pre-wrap', textAlign: 'center' }}>
                {acceptedFile.name}
              </div>
            </div>
          )) : (
            <div style={{ display: 'inline-block' }}>
              {FileIcon
                ? (
                  <div style={{ textAlign: 'center' }}>
                    <FileIcon height={48} width={48} />
                  </div>
                )
                : null
              }
              <div style={{ paddingTop: '10px', whiteSpace: 'pre-wrap', textAlign: 'center' }}>
                {isDragAccept && renderDetails(status.accepted)}
                {isDragReject && renderDetails(status.rejected)}
                {!isDragActive && renderDetails(status.ready)}
              </div>
            </div>
          )
        }
      </div>
    </div>
  );
}

const dropzoneStyles = {
  baseStyle: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
    height: '170px',
    alignItems: 'center',
    padding: '20px',
    borderWidth: 2,
    borderRadius: 2,
    borderColor: '#eeeeee',
    borderStyle: 'dashed',
    backgroundColor: '#fafafa',
    color: '#bdbdbd',
    outline: 'none',
    transition: 'border .24s ease-in-out',
  },
  activeStyle: {
    borderColor: '#2196f3',
  },

  acceptStyle: {
    borderColor: '#00e676',
  },

  rejectStyle: {
    borderColor: '#ff1744',
  },
};


Dropzone.propTypes = {
  accept: PropTypes.string.isRequired,
  saveFile: PropTypes.func,
  rejectFile: PropTypes.func,
  uploadHandler: PropTypes.func,
  icon: PropTypes.func,
  acceptedIcon: PropTypes.func,
  multiple: PropTypes.bool,
  disabled: PropTypes.bool,
  shouldUpload: PropTypes.bool,
  acceptStyle: PropTypes.object,
  activeStyle: PropTypes.object,
  rejectStyle: PropTypes.object,
  baseStyle: PropTypes.object,
  maxSize: PropTypes.number,
  minSize: PropTypes.number,
};

Dropzone.defaultProps = {
  saveFile: () => {},
  rejectFile: () => {},
  uploadHandler: () => {},
  icon: null,
  acceptedIcon: null,
  multiple: false,
  disabled: false,
  shouldUpload: false,
  acceptStyle: dropzoneStyles.acceptStyle,
  activeStyle: dropzoneStyles.activeStyle,
  rejectStyle: dropzoneStyles.rejectStyle,
  baseStyle: dropzoneStyles.baseStyle,
  maxSize: undefined,
  minSize: undefined,
  removeFile: false,
};
