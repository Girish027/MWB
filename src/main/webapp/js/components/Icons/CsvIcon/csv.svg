<?xml version="1.0" encoding="UTF-8"?>
<svg width="49px" height="49px" viewBox="0 0 49 49" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 54.1 (76490) - https://sketchapp.com -->
    <title>csv</title>
    <desc>Created with Sketch.</desc>
    <g id="Intent-and-Flow-Discovery" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="01B-Create-New-Project---Upload-File" transform="translate(-658.000000, -333.000000)" fill="#9B9B9B" fill-rule="nonzero">
            <g id="Group-4" transform="translate(445.000000, 155.000000)">
                <g id="Group-7">
                    <g id="Group-4" transform="translate(27.500000, 79.481906)">
                        <g id="Group-8" transform="translate(0.000000, 36.728489)">
                            <g id="Upload-file" transform="translate(0.000000, 28.100796)">
                                <g id="csv" transform="translate(186.000000, 33.909308)">
                                    <path d="M47.987,21.8794821 C47.981,21.7887248 47.964,21.7019569 47.934,21.6161863 C47.923,21.5842717 47.915,21.5533544 47.901,21.522437 C47.853,21.4187145 47.792,21.3209759 47.708,21.2381973 C47.707,21.2371999 47.707,21.2371999 47.707,21.2371999 L42,15.5444256 L42,9.97332579 C42,9.95138447 41.989,9.93243515 41.987,9.91049383 C41.981,9.82272857 41.964,9.7379553 41.936,9.65417936 C41.925,9.62226472 41.917,9.59134741 41.902,9.5604301 C41.853,9.45471284 41.792,9.35398225 41.706,9.26821165 L32.706,0.292218446 C32.62,0.206447844 32.519,0.144613224 32.412,0.0967412601 C32.382,0.082778604 32.352,0.0747999434 32.321,0.0648266176 C32.236,0.0349066402 32.149,0.0179519864 32.058,0.0129653235 C32.039,0.00997332579 32.021,0 32,0 L7,0 C6.448,0 6,0.446804995 6,0.997332579 L6,15.5444256 L0.293,21.2362026 C0.293,21.2362026 0.292,21.2371999 0.291,21.2381973 C0.207,21.3219732 0.147,21.4197118 0.099,21.522437 C0.085,21.5533544 0.077,21.5842717 0.066,21.6161863 C0.036,21.7019569 0.018,21.7887248 0.013,21.8794821 C0.011,21.9014234 0,21.9193754 0,21.9413167 L0,40.8906357 C0,41.4411633 0.448,41.8879683 1,41.8879683 L6,41.8879683 L6,46.8746312 C6,47.4251588 6.448,47.8719638 7,47.8719638 L41,47.8719638 C41.552,47.8719638 42,47.4251588 42,46.8746312 L42,41.8879683 L47,41.8879683 C47.552,41.8879683 48,41.4411633 48,40.8906357 L48,21.9413167 C48,21.9193754 47.989,21.9014234 47.987,21.8794821 Z M44.586,20.9439841 L42,20.9439841 L42,18.3648821 L44.586,20.9439841 Z M38.586,8.97599321 L33,8.97599321 L33,3.40489342 L38.586,8.97599321 Z M8,1.99466516 L31,1.99466516 L31,9.97332579 C31,10.5238534 31.448,10.9706584 32,10.9706584 L40,10.9706584 L40,15.9573213 L40,20.9439841 L8,20.9439841 L8,15.9573213 L8,1.99466516 Z M6,18.3648821 L6,20.9439841 L3.414,20.9439841 L6,18.3648821 Z M40,45.8772986 L8,45.8772986 L8,41.8879683 L40,41.8879683 L40,45.8772986 Z M46,39.8933031 L2,39.8933031 L2,22.9386493 L7,22.9386493 L41,22.9386493 L46,22.9386493 L46,39.8933031 Z" id="Shape"></path>
                                    <path d="M23.422,27.810619 C23.569,27.6749817 23.734,27.5762458 23.915,27.5134138 C24.096,27.4515792 24.283,27.4206619 24.476,27.4206619 C25.145,27.4206619 25.7,27.6859524 26.142,28.2175306 L27.264,26.7594304 C26.935,26.3754574 26.53,26.0842363 26.049,25.8857671 C25.567,25.6882952 25.021,25.5895593 24.409,25.5895593 C23.99,25.5895593 23.576,25.6603699 23.168,25.8009938 C22.76,25.942615 22.394,26.1600335 22.071,26.4542466 C21.748,26.7484598 21.487,27.1184701 21.289,27.5642778 C21.091,28.0110828 20.991,28.5456531 20.991,29.1669913 C20.991,29.6646602 21.056,30.0905212 21.186,30.446569 C21.316,30.803614 21.492,31.1137845 21.713,31.3790749 C21.934,31.6453627 22.189,31.8737519 22.478,32.0662371 C22.767,32.2587223 23.076,32.4332555 23.405,32.5918313 C23.926,32.8521351 24.357,33.1343803 24.697,33.439564 C25.037,33.7447478 25.207,34.1576435 25.207,34.6772538 C25.207,35.208832 25.065,35.6207304 24.782,35.9149435 C24.499,36.2091566 24.142,36.3557645 23.711,36.3557645 C23.326,36.3557645 22.949,36.2650072 22.58,36.08449 C22.211,35.9039728 21.897,35.6546397 21.636,35.3384853 L20.531,36.8304948 C20.871,37.2264358 21.324,37.5535609 21.891,37.8138647 C22.458,38.0741685 23.075,38.2038218 23.744,38.2038218 C24.209,38.2038218 24.651,38.1250325 25.07,37.9664566 C25.489,37.8078807 25.855,37.5735076 26.167,37.2623398 C26.479,36.9521694 26.728,36.570191 26.915,36.1183993 C27.102,35.6666077 27.196,35.1460001 27.196,34.5585712 C27.196,34.0499316 27.117,33.6091106 26.958,33.2361082 C26.799,32.8631058 26.595,32.5409674 26.346,32.2696929 C26.097,31.9984185 25.819,31.7670373 25.513,31.5745521 C25.207,31.3820669 24.907,31.2125204 24.612,31.0659125 C24.113,30.8175767 23.711,30.5542809 23.405,30.2770224 C23.099,30.0007613 22.946,29.6078123 22.946,29.0991726 C22.946,28.8049595 22.988,28.5506397 23.074,28.3362132 C23.159,28.1217867 23.275,27.9462562 23.422,27.810619 Z" id="Shape"></path>
                                    <path d="M15.177,28.7770342 C15.381,28.3810932 15.636,28.0679308 15.942,27.8355523 C16.248,27.6041711 16.605,27.4884805 17.013,27.4884805 C17.75,27.4884805 18.35,27.8215896 18.815,28.4888051 L19.988,27.0646142 C19.659,26.5898839 19.234,26.2258575 18.713,25.9705403 C18.192,25.7162205 17.585,25.5895593 16.894,25.5895593 C16.225,25.5895593 15.616,25.7451432 15.066,26.0553136 C14.516,26.3664814 14.049,26.8013184 13.663,27.360822 C13.278,27.9213229 12.98,28.587541 12.771,29.3614711 C12.561,30.1363985 12.457,30.9811392 12.457,31.8966905 C12.457,32.8122418 12.562,33.6539905 12.771,34.4229339 C12.98,35.1918774 13.275,35.8551035 13.655,36.4146071 C14.035,36.975108 14.499,37.412937 15.049,37.7290914 C15.599,38.0452459 16.214,38.2038218 16.894,38.2038218 C17.642,38.2038218 18.262,38.0572139 18.755,37.7630008 C19.248,37.4687876 19.653,37.0848146 19.97,36.6100843 L18.797,35.2367573 C18.604,35.5309704 18.363,35.7793063 18.075,35.9827621 C17.787,36.1862179 17.443,36.2879459 17.047,36.2879459 C16.628,36.2879459 16.262,36.1722553 15.95,35.9398768 C15.638,35.7084956 15.381,35.3953332 15.176,34.9993922 C14.971,34.6034511 14.819,34.1376968 14.717,33.6001346 C14.615,33.0635696 14.564,32.4960874 14.564,31.8966905 C14.564,31.286323 14.615,30.7128568 14.717,30.1752945 C14.82,29.6387296 14.973,29.1729753 15.177,28.7770342 Z" id="Shape"></path>
                                    <polygon id="Shape" points="32.007 35.5249864 31.956 35.5249864 29.95 25.8947431 27.672 25.8947431 30.613 37.898638 33.197 37.898638 36.189 25.8947431 34.013 25.8947431"></polygon>
                                </g>
                            </g>
                        </g>
                    </g>
                </g>
            </g>
        </g>
    </g>
</svg>