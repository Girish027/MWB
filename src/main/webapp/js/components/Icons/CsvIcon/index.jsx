import React from 'react';
import PropTypes from 'prop-types';

const CsvIcon = (props) => {
  const {
    height, width, fill, ...otherProps
  } = props;

  return (
    <svg width={width} height={height} viewBox="0 0 49 49" {...otherProps}>
      <g fill={fill} fillRule="nonzero">
        <path d="M48.487 22.1a.986.986 0 0 0-.053-.263c-.011-.032-.019-.063-.033-.094a.955.955 0 0 0-.193-.284l-.001-.001-5.707-5.693v-5.571c0-.022-.011-.041-.013-.063a1.023 1.023 0 0 0-.051-.256c-.011-.032-.019-.063-.034-.094a.994.994 0 0 0-.196-.292l-9-8.976a.97.97 0 0 0-.294-.196c-.03-.014-.06-.022-.091-.032a.94.94 0 0 0-.263-.052C32.539.23 32.521.22 32.5.22h-25c-.552 0-1 .447-1 .998v14.547L.793 21.457l-.002.002a.977.977 0 0 0-.192.284c-.014.03-.022.062-.033.094a.948.948 0 0 0-.053.263c-.002.022-.013.04-.013.062v18.95c0 .55.448.996 1 .996h5v4.987c0 .55.448.997 1 .997h34c.552 0 1-.446 1-.997v-4.987h5c.552 0 1-.446 1-.997v-18.95c0-.021-.011-.04-.013-.061zm-3.401-.936H42.5v-2.579l2.586 2.58zm-6-11.968H33.5v-5.57l5.586 5.57zM8.5 2.216h23v7.978c0 .55.448.997 1 .997h8V21.164h-32V2.215zm-2 16.37v2.578H3.914L6.5 18.585zm34 27.512h-32v-3.99h32v3.99zm6-5.984h-44V23.159h44v16.955z" />
        <path d="M23.922 28.031c.147-.136.312-.234.493-.297.181-.062.368-.093.561-.093.669 0 1.224.265 1.666.797l1.122-1.458a3.164 3.164 0 0 0-1.215-.874c-.482-.197-1.028-.296-1.64-.296-.419 0-.833.07-1.241.211a3.145 3.145 0 0 0-1.097.654 3.276 3.276 0 0 0-.782 1.11c-.198.447-.298.981-.298 1.602 0 .498.065.924.195 1.28.13.357.306.667.527.933.221.266.476.494.765.687.289.192.598.367.927.525.521.26.952.543 1.292.848.34.305.51.718.51 1.238 0 .531-.142.943-.425 1.237-.283.295-.64.441-1.071.441-.385 0-.762-.09-1.131-.271a2.816 2.816 0 0 1-.944-.746l-1.105 1.492c.34.396.793.723 1.36.983.567.26 1.184.39 1.853.39.465 0 .907-.078 1.326-.237a3.077 3.077 0 0 0 1.097-.704 3.45 3.45 0 0 0 .748-1.144c.187-.452.281-.973.281-1.56 0-.509-.079-.95-.238-1.322a3.366 3.366 0 0 0-.612-.967 4.063 4.063 0 0 0-.833-.695 10.331 10.331 0 0 0-.901-.509c-.499-.248-.901-.511-1.207-.788-.306-.277-.459-.67-.459-1.178 0-.295.042-.549.128-.763.085-.215.201-.39.348-.526zM15.677 28.998c.204-.396.459-.71.765-.942a1.726 1.726 0 0 1 1.071-.347c.737 0 1.337.333 1.802 1l1.173-1.424a3.282 3.282 0 0 0-1.275-1.094c-.521-.254-1.128-.38-1.819-.38-.669 0-1.278.155-1.828.465a4.224 4.224 0 0 0-1.403 1.305c-.385.56-.683 1.227-.892 2.001a9.67 9.67 0 0 0-.314 2.535c0 .916.105 1.757.314 2.526.209.77.504 1.433.884 1.992.38.56.844.998 1.394 1.315.55.316 1.165.474 1.845.474.748 0 1.368-.146 1.861-.44a3.757 3.757 0 0 0 1.215-1.153l-1.173-1.374a2.76 2.76 0 0 1-.722.746 1.738 1.738 0 0 1-1.028.305c-.419 0-.785-.115-1.097-.348a2.674 2.674 0 0 1-.774-.94 5.103 5.103 0 0 1-.459-1.4 9.11 9.11 0 0 1-.153-1.703c0-.61.051-1.184.153-1.721a5.175 5.175 0 0 1 .46-1.398zM32.507 35.745h-.051l-2.006-9.63h-2.278l2.941 12.004h2.584l2.992-12.004h-2.176z" />
      </g>
    </svg>
  );
};

CsvIcon.defaultProps = {
  height: 49,
  width: 49,
  fill: '#9B9B9B',
};

CsvIcon.propTypes = {
  height: PropTypes.number,
  width: PropTypes.number,
  fill: PropTypes.string,
};

export default CsvIcon;
