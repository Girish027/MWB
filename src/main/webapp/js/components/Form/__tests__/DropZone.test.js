import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Doc } from '@tfs/ui-components';
import Dropzone from 'components/Form/Dropzone';
import CsvIcon from 'components/Icons/CsvIcon';

describe('<Dropzone />', () => {
  let wrapper;

  const props = {
    accept: 'text/csv, application/vnd.ms-excel, .csv',
    icon: { CsvIcon },
    acceptedIcon: { Doc },
    maxSize: 10,
    minSize: 5,
    multiple: false,
    saveFile: jest.fn(),
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<Dropzone
    {...propsObj}
  />);

  const getMountWrapper = (propsObj) => mount(<Dropzone
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getMountWrapper();
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly with default props', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
