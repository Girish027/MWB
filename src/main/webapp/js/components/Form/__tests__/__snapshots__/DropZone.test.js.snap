// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropzone /> Snapshots renders correctly with default props 1`] = `
<div
  onBlur={[Function]}
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDrop={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fafafa",
      "borderColor": "#eeeeee",
      "borderRadius": 2,
      "borderStyle": "dashed",
      "borderWidth": 2,
      "color": "#bdbdbd",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "height": "170px",
      "outline": "none",
      "padding": "20px",
      "transition": "border .24s ease-in-out",
    }
  }
  tabIndex={0}
>
  <input
    accept="text/csv, application/vnd.ms-excel, .csv"
    autoComplete="off"
    disabled={false}
    multiple={false}
    onChange={[Function]}
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />
  <div
    className="dropzone-content"
    style={
      Object {
        "margin": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <[object Object]
          height={48}
          width={48}
        />
      </div>
      <div
        style={
          Object {
            "paddingTop": "10px",
            "textAlign": "center",
            "whiteSpace": "pre-wrap",
          }
        }
      >
        <div>
          Drag and drop your file
          <div>
             or 
          </div>
          <div
            style={
              Object {
                "color": "#004c97",
                "cursor": "pointer",
              }
            }
          >
             BROWSE
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
